### DISCUSSION TITLE
config project w/o modules

### DISCUSSION THREAD

**Discussion by BradWalker** [eugene-tarassov/vivado-risc-v#175] [Q&A] [answered]
I've been looking at this project for the past week and have to say it's a rather well engineered RISC-V project.

What I'm thinking of doing is using Vivado-Risc-V project as a basis for doing some OS research. I want to remove the Enet & SD support. Basically just want to use a single RISC-V core w/ a UART.

From looking at the TCL scripts, it should be fairly "straight-forward" to do this. Would this be your expectation also?

In addition, any plans on parameterizing the project to make addition/removal of modules "easier" in the future?

Thanks for any insight provided.

**Comment by eugene-tarassov** [comment, marked as answer]
Yes, it is easy to remove or add modules to the project. You can either use Vivado GUI to to modify the block design and save it as Tcl file, or edit Tcl files in a text editor.

At the moment, I don't have plans to make addition/removal of modules easier. I think that Vivado Tcl is already a flexible and easy to use scripting tool, which allows quick project modifications by editing a script.

**Reply by BradWalker** [reply]
Hey @eugene-tarassov , thanks for the response. It's what I would have expected. And yes, I do agree editing the Tcl scripts is pretty easy.
